lib/dal/src/func/associations.rs

```diff
@@ -166,8 +82,28 @@ impl FuncAssociations {
         let (associations, input_type) = match func.kind {
             FuncKind::Action => {
                 let schema_variant_ids = SchemaVariant::list_for_action_func(ctx, func.id).await?;
+                let action_prototype_ids =
+                    DeprecatedActionPrototype::list_for_func_id(ctx, func.id).await?;
+
+                // TODO(nick): right now, we just grab the first one and it decides the action kind for all of them.
+                // This should be configurable on a "per prototype" basis in the future.
+                let kind = match action_prototype_ids.first() {
+                    Some(action_prototype_id) => {
+                        let action_prototype = DeprecatedActionPrototype::get_by_id_or_error(
+                            ctx,
+                            *action_prototype_id,
+                        )
+                        .await?;
+                        action_prototype.kind
+                    }
+                    None => DeprecatedActionKind::Create,
+                };
```

Comment:
I don't love this because different action prototypes can technically have different action kinds for the same func. However, this is how it worked in the old engine and we are in the middle of an actions re-write, so restoring old behavior is fine for now.